extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -94,8 +96,9 @@ public OpenTelemetryMetricsProtobufReader(
           Map<String, Object> resourceAttributes = resourceMetrics.getResource()
               .getAttributesList()
               .stream()
-              .collect(Collectors.toMap(kv -> resourceAttributePrefix + kv.getKey(),
-                  kv -> parseAnyValue(kv.getValue())));
+              .collect(HashMap::new,
```

Comment:
any reason we are not using Collectors.toMap anymore?

Reply:
Since the function `parseAnyValue` is nullable, it could not be used in the Collectors.toMap lambda function. This is a workaround.

Reply:
you might be able to add a predicate in there before calling the collector instead

Reply:
@xvrl How do you suggest we do this. One way I can think of would be to have an immutable set and filter it. so

```java
  private final Set<AnyValue.ValueCase> supportedValueCases = ImmutableSet.of(
      AnyValue.ValueCase.INT_VALUE,
      AnyValue.ValueCase.STRING_VALUE,
      AnyValue.ValueCase.BOOL_VALUE,
      AnyValue.ValueCase.DOUBLE_VALUE
  );

// ...
.stream()
.filter(kv -> supportedValueCases.contains(kv.getValue().getValueCase()))
.collect(Collections.toMap(resourceAttributePrefix + kv.getKey(), parseAnyValue(kv.getValue()))

// ...
  @Nullable
  private static Object parseAnyValue(AnyValue value)
  {
    switch (value.getValueCase()) {
      case INT_VALUE:
        return value.getIntValue();
      case BOOL_VALUE:
        return value.getBoolValue();
      case DOUBLE_VALUE:
        return value.getDoubleValue();
      case STRING_VALUE:
        return value.getStringValue();

      // TODO: Support KVLIST_VALUE, ARRAY_VALUE and BYTES_VALUE
      default:
        // This should never be reached
        return null;
    }
  }
```

Reply:
might be more complicated than necessary